common/thorhelper/thorxmlwrite.cpp

```diff
@@ -596,6 +620,24 @@ void CommonJsonWriter::outputEndDataset(const char *dsname)
         outputEndNested(dsname);
 }
 
+void CommonJsonWriter::cutFrom(IInterface *location, StringBuffer& databuf)
+{
+    if (flusher)
+        throwUnexpected();
+
+    CommonXmlPosition *position = dynamic_cast<CommonXmlPosition *>(location);
+    if (!position)
+        return;
+    if (position->pos < out.length())
+    {
+        databuf.append(out.length() - position->pos, out.str() + position->pos);
+        out.setLength(position->pos);
+        needDelimiter = position->needDelimiter;
+        indent = position->indent;
+        nestLimit = position->nestLimit;
+    }
```

Comment:
Is the above code identical to that in CommonXmlWriter?
Should be commoned up iso, maybe via a base class?